internal/cluster/clusterworkflow/activity_create_node_pool_label_set.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterworkflow
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/banzaicloud/pipeline/internal/cluster"
+	"github.com/banzaicloud/pipeline/internal/cluster/distribution"
+	"github.com/banzaicloud/pipeline/pkg/cadence"
+	"github.com/banzaicloud/pipeline/pkg/kubernetes/custom/npls"
+	"github.com/banzaicloud/pipeline/pkg/providers"
+	"github.com/banzaicloud/pipeline/src/cluster/nodelabels"
+)
+
+const CreateNodePoolLabelSetActivityName = "create-node-pool-label-set"
+
+type CreateNodePoolLabelSetActivity struct {
+	clusters             cluster.Store
+	dynamicClientFactory DynamicClientFactory
+	namespace            string
+}
+
+// NewCreateNodePoolLabelSetActivity returns a new CreateNodePoolLabelSetActivity.
+func NewCreateNodePoolLabelSetActivity(
+	clusters cluster.Store,
+	dynamicClientFactory DynamicClientFactory,
+	namespace string,
+) CreateNodePoolLabelSetActivity {
+	return CreateNodePoolLabelSetActivity{
+		clusters:             clusters,
+		dynamicClientFactory: dynamicClientFactory,
+		namespace:            namespace,
+	}
+}
+
+type CreateNodePoolLabelSetActivityInput struct {
+	ClusterID   uint
+	RawNodePool cluster.NewRawNodePool
+}
+
+func (a CreateNodePoolLabelSetActivity) Execute(ctx context.Context, input CreateNodePoolLabelSetActivityInput) error {
+	client, err := a.dynamicClientFactory.FromClusterID(ctx, input.ClusterID)
+	if err != nil {
+		return cadence.WrapClientError(err)
+	}
+
+	c, err := a.clusters.GetCluster(ctx, input.ClusterID)
+	if err != nil {
+		return cadence.WrapClientError(err)
+	}
+
+	var name string
+	var labels map[string]string
+
+	switch {
+	case c.Cloud == providers.Amazon && c.Distribution == "eks":
```

Comment:
```suggestion
	case c.Cloud == providers.Amazon && c.Distribution == cluster.EKS:
```

Reply:
I don't want to import that package here for now.